### ISSUE TITLE
[3] User Enumeration on frontend "Forgot your password"

### ISSUE THREAD

**Issue by PhilETaylor** [joomla/joomla-cms#31892]
### Steps to reproduce the issue

Use "Forgot your password" on frontend of Joomla 4 beta 6 
enter a valid syntax email of a non-user Eg : No-Such-Email@example.com
Click Submit

### Expected result

"If a user matched that email address then we sent an email to them"

### Actual result

"Reset password failed: Invalid email address" because no user matched that email address

<img width="452" alt="Screenshot 2021-01-08 at 14 18 19" src="https://user-images.githubusercontent.com/400092/104025337-5f546500-51bc-11eb-91d6-07b99f8bb2a6.png">

### Additional comments

Note that if a user EXISTS with the email address provided then the page redirects to 
<img width="903" alt="Screenshot 2021-01-08 at 14 20 01" src="https://user-images.githubusercontent.com/400092/104025476-9aef2f00-51bc-11eb-8682-1bb5a2f6fa16.png">

**Comment by wilsonge** [comment]
This code is line for line the same as j3. I haven't tested but struggle to believe this is j4 specific

**Comment by PhilETaylor** [comment]
I was asked to test Joomla 4 beta 6 - so I did and reported what I found.

As the JSST are so slow to respond to any security issue reported, Im not surprised. The last time I asked for an update (October 2020) I was told that TEN of my reports were still outstanding and unresolved. 

<img width="375" alt="Screenshot 2021-01-08 at 16 01 56" src="https://user-images.githubusercontent.com/400092/104036090-e577a800-51ca-11eb-922a-61e7f28b1726.png">
